docs/decisions/0004-technology-selection.rst

```diff
@@ -0,0 +1,103 @@
+0004: Authorization Technology Selection
+#########################################
+
+Status
+******
+
+**Draft** *2025-09-19*
+
+Context
+*******
+
+Authorization is a common challenge across software platforms, and many open-source communities have already built frameworks to address it. To understand what could work for Open edX, we reviewed a range of existing technologies and compared them against `a consistent set of evaluation criteria`_, to:
+
+* Avoid reinventing the wheel by leveraging proven approaches.
+* Learn from established patterns (RBAC, ABAC, ReBAC, policy-based models).
+* Choose solutions that balance flexibility with maintainability.
+* Ensure long-term scalability and alignment with modern best practices.
+
+We analyzed a variety of authorization technologies, including Django permissions, Django-guardian, Django-prbac, Bridgekeeper, Edx-rbac, Casbin, Spicedb, Keycloak, Cerbos/Permguard. The complete analysis of these solutions can be found here: `Authorization Technologies Reviewed`_.
+
+Following this preliminary assessment, we determined that Django-prbac, Casbin, and OpenFGA were the solutions most closely aligned with the requirements. We create a comparison table to decide which one is more suitable for our use case, taking into account factors such as integration fit, permission management, performance, extensibility, maturity, security, learning curve, total cost of operations, and other relevant considerations. Here you can find the `Authorization Technologies Comparison Table`_.
+
+Decision
+********
+
+* We choose to use `Casbin`_ as the authorization engine for Open edX, which allows us to have a robust foundation, enabling long-term evolution; improves security because it enforce the principle of least privilege; help us maintain a centrilize logic, making it easier to manage and update; and promotes best practice because has support for multiple well-understood authorization models. This decision of using Casbin is also aligned with the `Authorization Model Foundations ADR`_.
+
+* We'll use the production-ready Python (pycasbin) library and Django integration (django-authorization) for Django native APIs.
```

Comment:
As far as I understand, we dropped the use of `django-authorization` because it wasn't viable for our use case and scale. Here's more details about our initial discussion: https://github.com/openedx/openedx-authz/pull/37#discussion_r2336276117

Reply:
Thanks for that! Fixed!